src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/PackageLoader.cs

```diff
@@ -265,21 +266,29 @@ private async Task<IEnumerable<NuGetFramework>> GetTargetFrameworksAsync(Package
             foreach (var source in _packageSources.Value)
             {
                 var repo = GetSourceRepository(source);
-                var packageFinder = await repo.GetResourceAsync<FindPackageByIdResource>(token).ConfigureAwait(false);
-                if (await packageFinder.DoesPackageExistAsync(packageReference.Name, packageVersion, _cache, _nugetLogger, token).ConfigureAwait(false))
+                try
                 {
-                    var memoryStream = new MemoryStream();
-                    if (await packageFinder.CopyNupkgToStreamAsync(packageReference.Name, packageVersion, memoryStream, _cache, _nugetLogger, token).ConfigureAwait(false))
-                    {
-                        _logger.LogDebug("Package {NuGetPackage} download from feed {NuGetFeed}", packageReference, source.Source);
-                        return new PackageArchiveReader(memoryStream, false);
-                    }
-                    else
+                    var packageFinder = await repo.GetResourceAsync<FindPackageByIdResource>(token).ConfigureAwait(false);
+                    if (await packageFinder.DoesPackageExistAsync(packageReference.Name, packageVersion, _cache, _nugetLogger, token).ConfigureAwait(false))
                     {
-                        _logger.LogDebug("Failed to download package {NuGetPackage} from feed {NuGetFeed}", packageReference, source.Source);
-                        memoryStream.Close();
+                        var memoryStream = new MemoryStream();
```

Comment:
don't our code analysis rules balk at this? was expecting to see a using stmt

Reply:
oh, i was expecting this to be a temporary buffer. I see we're giving it away to *PackageArchiveReader* now